src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -215,9 +215,12 @@ internal void AddProject(AbstractProject project)
 
         internal void StartPushingToWorkspaceAndNotifyOfOpenDocuments(IEnumerable<AbstractProject> projects)
         {
-            foreach (var hostState in _workspaceHosts)
+            using (Dispatcher.CurrentDispatcher.DisableProcessing())
```

Comment:
Should this just be updated to call `NotifyWorkspaceHosts` that already has this check? It also has the "workspace host ready" check which I think it does need...?